src/components/AssetDetailComponent.jsx

```diff
@@ -31,10 +31,7 @@ export class AssetDetailComponent extends Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    if (
-      this.props.hasError &&
-      (this.props.errorMessage === nextProps.errorMessage)
-    ) {
+    if (this.props.hasError && (this.props.errorMessage === nextProps.errorMessage)) {
```

Comment:
The shouldComponentUpdate logic can be simplified to:
```
shouldComponentUpdate(nextProps) {
  return this.props.hasError && (this.props.errorMessage !== nextProps.errorMessage);
}
```

Please test if the change works as expected

Reply:
The changes work as expected